### PR TITLE
add upgrade tests in query package for all version combinations

### PR DESCRIPTION
**Pull request by mangalaman93**
This PR allows us to run upgrade tests for various combinations of before and after dgraph versions in the query package.

### CODE REVIEW

File: query/upgrade_test.go [hunk 1]
```diff
@@ -27,36 +28,70 @@ import (
...
+		{"v21.03.0-92-g0c9f60156", "v23.0.0-beta1"},
+		// {"v21.03.0-96-g65fff46c4-slash", "v23.0.0-beta1"},
+		// {"v21.03.0-98-g19f71a78a-slash", "v23.0.0-beta1"},
+		// {"v21.03.0-99-g4a03c144a-slash", "v23.0.0-beta1"},
+		// {"v21.03.1", "v23.0.0-beta1"},
+		// {"v21.03.2", "v23.0.0-beta1"},
+		// {"v21.12.0", "v23.0.0-beta1"},
+		// {"v22.0.0", "v23.0.0-beta1"},
+		{"v22.0.1", "v23.0.0-beta1"},
+		// {"v22.0.2", "v23.0.0-beta1"},
```
Comment (joshua-goldstein) [inline query/upgrade_test.go:78]: we might want to paralellize this in the future, perhaps by making this configurable via a command line argument to `go test`

File: query/upgrade_test.go [hunk 2]
```diff
@@ -27,36 +28,70 @@ import (
...
+	}
+	for _, cv := range comboVersions {
+		log.Printf("running: backup in [%v], restore in [%v]", cv.before, cv.after)
```
Comment (joshua-goldstein) [inline query/upgrade_test.go:81]: can add if/else logic here based on arguments provided

File: query/upgrade_test.go (hunk 1 above)
Comment (harshil-goel) [reply query/upgrade_test.go:78]: We can just run the official go test using m.Run("name"). If we use proper names, go test will run a specific test only.

File: query/upgrade_test.go [hunk 3]
```diff
@@ -27,36 +28,70 @@ import (
...
+		runTest(cv.before, cv.after)
+	}
 }
+
+// userClient, err := testutil.DgraphClient("0.0.0.0:33196")
```
Comment (harshil-goel) [inline query/upgrade_test.go:86]: Does this need to be removed?

**Review by harshil-goel — APPROVED** [review]
This is really great to see. The test could be written in minutes after the new framework. Lets write more tests now!!

File: query/upgrade_test.go (hunk 1 above)
Comment (mangalaman93) [reply query/upgrade_test.go:78]: no, that works when we are inside a Test not in TestMain

File: query/upgrade_test.go (hunk 3 above)
Comment (mangalaman93) [reply query/upgrade_test.go:86]: wip, I will fix it.

**Review by harshil-goel — APPROVED** [review]

**Review by anurags92 — APPROVED** [review]

**Pull request merged**